lib/errors/database-error.ts

```diff
@@ -0,0 +1,40 @@
+import BaseError, { ErrorOptions } from './base-error';
+
+interface IDatabaseError extends Error {
```

Comment:
Unsure about the name of this interface, but might reuse and rename it elsewhere when necessary

Reply:
Just seen the `CommonErrorProperties` interface in `types/lib/errors.d.ts`, will change it then